openmdao/core/tests/test_problem.py

```diff
@@ -1008,6 +1008,25 @@ def test_feature_get_set_array_with_units(self):
         assert_rel_error(self, prob.get_val('comp.x'), np.array([1.0, 5.0]), 1e-6)
         assert_rel_error(self, prob.get_val('comp.x', 'm', indices=1), 5.0e-2, 1e-6)
 
+    def test_feature_get_set_array_with_slicer(self):
+        import numpy as np
+        import openmdao.api as om
+
+        prob = om.Problem()
+        prob.model.add_subsystem('comp', om.ExecComp('y=x+1.',
+                                                     x={'value': np.array([[1., 2.], [3., 4.]]),},
+                                                     y={'shape': (2, 2), }))
+
+        prob.setup()
+        prob.run_model()
+
+        assert_rel_error(self, prob.get_val('comp.x', indices=om.slicer[:,0]), [1., 3.], 1e-6)
+
+        prob.set_val('comp.x', [5., 6.], indices=om.slicer[:,0])
+        assert_rel_error(self, prob.get_val('comp.x', indices=om.slicer[:,0]), [5., 6.], 1e-6)
+        prob.run_model()
+        assert_rel_error(self, prob.get_val('comp.y', indices=om.slicer[:,0]), [6., 7.], 1e-6)
```

Comment:
I wonder if we should show an example for a single index get too, like [0, 1]. Just to drive home the point that all of the normal slicing syntax is valid.  Could also show that it works with a -1.